src/test/java/com/exasol/adapter/dialects/ScalarFunctionsAbstractIT.java

```diff
@@ -0,0 +1,495 @@
+        return CellMatcherFactory.cellMatcher(object, TypeMatchMode.NO_JAVA_TYPE_CHECK, BigDecimal.valueOf(0.0001));
+    }
+
+    private Stream<Arguments> getScalarFunctionWithNoParameters() {
+        return FUNCTIONS_WITH_NO_PARENTHESIS.stream().map(Arguments::of);
+    }
+
+    /**
+     * This test tests functions that do not have parenthesis (e.g. CURRENT_SCHEMA).
+     * <p>
+     * Since the result of all of this functions is different on different databases or at different time, we just test
+     * that they don't throw an exception on the Virtual Schema.
+     *
+     * </p>
+     *
+     * @param function function to test.
+     */
+    @ParameterizedTest
+    @MethodSource("getScalarFunctionWithNoParameters")
+    void testFunctionsWithNoParenthesis(final ScalarFunctionCapability function) {
+        runOnExasol(statement -> {
+            assertDoesNotThrow(() -> statement.executeQuery(getVirtualSchemaQuery(getFunctionName(function))).close());
+        });
+    }
+
+    /**
+     * For the random function it obviously makes no sense to check if the output is the same on two runs. For that
+     * reason we check here if it produces at least some different results.
+     */
+    @Test
+    void testRand() {
+        runOnExasol(statement -> {
+            final String query = getVirtualSchemaQuery("RAND()");
+            try (final ResultSet result = statement.executeQuery(query)) {
+                result.next();
+                final double value = result.getDouble(1);
+                assertThat(Double.isFinite(value), equalTo(true));
+            }
+        });
+    }
+
+    @Test
+    void testSysGUID() {
+        runOnExasol(statement -> {
+            assertDoesNotThrow(() -> statement.executeQuery(getVirtualSchemaQuery("SYS_GUID()")).close());
+        });
+    }
+
+    @Test
+    void testAdd() {
+        runOnExasol(statement -> {
+            try (final ResultSet result = statement.executeQuery(getVirtualSchemaQuery("1 + 3"))) {
+                assertThat(result, table().row(4).matches(TypeMatchMode.NO_JAVA_TYPE_CHECK));
+            }
+        });
+    }
+
+    @Test
+    void testSub() {
+        runOnExasol(statement -> {
+            try (final ResultSet result = statement.executeQuery(getVirtualSchemaQuery("3 - 1"))) {
+                assertThat(result, table().row(2).matches(TypeMatchMode.NO_JAVA_TYPE_CHECK));
+            }
+        });
+    }
+
+    @Test
+    void testDiv() {
+        runOnExasol(statement -> {
+            try (final ResultSet result = statement.executeQuery(getVirtualSchemaQuery("3 / 2"))) {
+                assertThat(result, table().row(1.5).matches(TypeMatchMode.NO_JAVA_TYPE_CHECK));
+            }
+        });
+    }
+
+    @Test
+    void testMult() {
+        runOnExasol(statement -> {
+            try (final ResultSet result = statement.executeQuery(getVirtualSchemaQuery("4 * 2"))) {
+                assertThat(result, table().row(8).matches(TypeMatchMode.NO_JAVA_TYPE_CHECK));
```

Comment:
These four tests can be one parameterized.

Reply:
just realized that these tests are still broken. So I'll create a PR to this branch with a fix. Let's finish this review first.